CodeEdit/Features/NavigatorArea/SourceControlNavigator/Views/Changes/SourceControlNavigatorChangedFileView.swift

```diff
@@ -7,62 +7,104 @@
-                     changedFile.showInFinder(workspaceURL: workspaceURL)
-                 }
-                 Button("Reveal in Project Navigator") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Divider()
-             }
-             Group {
-                 Button("Open in New Tab") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Button("Open in New Window") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Button("Open with External Editor") {}
-                     .disabled(true) // TODO: Implementation Needed
-             }
-             Group {
-                 Divider()
-                 Button("Commit \(changedFile.fileName)...") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Divider()
-                 Button("Discard Changes in \(changedFile.fileName)...") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Divider()
-             }
-             Group {
-                 Button("Add \(changedFile.fileName)") {}
-                     .disabled(true) // TODO: Implementation Needed
-                 Button("Mark \(changedFile.fileName) as Resolved") {}
-                     .disabled(true) // TODO: Implementation Needed
-             }
-         }
-         .padding(.leading, 15)
-     }
- }
+struct SourceControlNavigatorChangedFileView: View {
+    @ObservedObject var sourceControlManager: SourceControlManager
+    var changedFile: CEWorkspaceFile
+
+    var folder: String? {
+        let rootPath = sourceControlManager.gitClient.directoryURL.relativePath
+        let filePath = changedFile.url.relativePath
+
+        // Should not happen, but just in case
+        if !filePath.hasPrefix(rootPath) {
+            return nil
+        }
+
+        let relativePath = filePath
+            .dropFirst(rootPath.count + 1) // Drop root folder
+            .dropLast(changedFile.name.count + 1) // Drop file name
+        return relativePath.isEmpty ? nil : String(relativePath)
+    }
+
+    var body: some View {
+        HStack(spacing: 5) {
+            Toggle("", isOn: .init(get: getSelectedFileState, set: setSelectedFile))
+                .labelsHidden()
+
+            HStack(spacing: 5) {
+                Image(systemName: changedFile.systemImage)
+                    .frame(width: 22)
+                    .foregroundColor(changedFile.iconColor)
+
+                Text(changedFile.name)
+                    .font(.system(size: 13))
+
+                if let folder {
+                    Text(folder)
+                        .font(.system(size: 11))
+                        .foregroundStyle(.secondary)
+                }
+
+                Spacer()
+
+                Text(changedFile.gitStatus?.description ?? "")
+                    .font(.system(size: 13))
+                    .foregroundColor(.secondary)
+            }
```

Comment:
This, combined with the other suggestion in this file makes a better layout that prefers displaying the file name when there's not room to display both the folder and the filename.
```suggestion
            HStack(alignment: .labelAlignmentGuide, spacing: 5) {
                Image(systemName: changedFile.systemImage)
                    .frame(width: 22)
                    .foregroundColor(changedFile.iconColor)
                    .layoutPriority(2)

                Text(changedFile.name)
                    .font(.system(size: 13))
                    .truncationMode(.tail)
                    .layoutPriority(1)
                    .alignmentGuide(.labelAlignmentGuide) { context in
                        context[.firstTextBaseline]
                    }

                if let folder {
                    GeometryReader { proxy in
                        if proxy.size.width > 10 {
                            HStack(alignment: .labelAlignmentGuide) {
                                Text(folder)
                                    .font(.system(size: 13))
                                    .foregroundStyle(.tertiary)
                                    .truncationMode(.head)
                                    .layoutPriority(0.5)
                                    .alignmentGuide(.labelAlignmentGuide) { context in
                                        context[.firstTextBaseline]
                                    }
                            }
                        } else {
                            EmptyView()
                        }
                    }
                }

                Spacer(minLength: 0)

                Text(changedFile.gitStatus?.description ?? "")
                    .font(.system(size: 13))
                    .foregroundColor(.secondary)
                    .layoutPriority(2)
            }
```

Reply:
@thecoolwinter Alignment is broken for me with this change

<img width="322" alt="Screenshot 2023-10-29 at 1 19 30 PM" src="https://github.com/CodeEditApp/CodeEdit/assets/42622715/3ac8cbaa-1583-4a7c-a7ae-89b4b7bb5b08">

Reply:
Hm, looks like it's an issue when the text doesn't fill the width. That's gonna be a larger change so we'll do it in a separate issue.